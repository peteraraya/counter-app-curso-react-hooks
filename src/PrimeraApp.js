// Es un componente UpperCamelCase 

// En  React tenemos dos tipos de componentes : los que estan basados en clases y los que estan basados en funciones


//  **** React estÃ¡ impulsando a componentes  basados en funciones *** 

// imr + tab : importa automaticamente 
import React, { Fragment } from 'react';

// Importar prop-types
import propTypes from 'prop-types';

const PrimeraApp = ({
                    saludo, 
                    subtitulo
                }) =>{

    return (
        <>
            <h2> { saludo }</h2>
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: propTypes.string.isRequired 
    // el saludo tiene que ser un string , con isRequired se hace obligatorio
}

// DefaultProps
PrimeraApp.defaultProps  ={
    subtitulo: 'Soy un subtitulo'
}
export default PrimeraApp;


// **** NOTAS ******

// Tener cuidado al hacer enter 
// se debe usar una etiqueta para envolver todo el codigo jsx , se envuelve entre parentesis
// en caso de mas elementos se utiliza un div o un <Fragment> el cual envuelve todo el codigo que desemos mostrar </Fragment>
// hay otra alternativa para los Fragment , dejando <> es un Frament mas optimo </>
// { } dentro de las llaves podemos enviar cualquier valor 