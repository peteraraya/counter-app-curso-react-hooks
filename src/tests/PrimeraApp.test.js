import React from 'react';
// import { render } from '@testing-library/react';

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas de componente <PrimeraApp />', () => {

  // test('<PrimeraApp /> debe mostrar el mensaje "Hola, Soy Goku"', () => {

  //   const saludo = "Hola, Soy Goku";
  //   // uso de wrapper
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);

  //   expect(getByText(saludo)).toBeInTheDocument();
  
  // });

  test('debe de mostrar <PrimeraApp /> correctamente', () => {
    // similar al render pero con otras opciones

    const saludo = "Hola, Soy Goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    // esperamos que nuestro componente se renderize de manera correcta
    expect( wrapper ).toMatchSnapshot();

  });

  test('debe de enviar el subtitulo enviado por props', () => {
    
      const saludo = 'Hola, Soy Goku';
      const subtitulo = 'Soy un subtitulo'
      const wrapper = shallow(
          <PrimeraApp 
            saludo={saludo} 
            subtitulo={subtitulo}
          />
          );

      // revisar parrafo - como hay uno 
      const textoParrafo = wrapper.find('p').text();

      expect(textoParrafo).toBe(subtitulo);

  });
  
  
  
  
})
