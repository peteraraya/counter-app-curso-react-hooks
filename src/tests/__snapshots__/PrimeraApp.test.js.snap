// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h2>
     
    Hola, Soy Goku
  </h2>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
